### DISCUSSION TITLE
Continuous time continuous learning

### DISCUSSION THREAD

**Discussion by knyazer** [DelftMercurians/cotix#47] [Ideas]
Train a function that represents how the loss function changes with parameter modifications. The problem in continuum learning is that your target 'drifts away', so effectively optimizers stop working: our loss is something in the lines of L(psi(theta), theta), where psi(theta) represents a 'task': given current model parameters, it outputs a 'loss' function, L is simply an alias for 'L(a, b) = a(b)' and theta is the model parameters. When we do a gradient descent on a batch of data, we compute grad L(psi(theta1), theta2) wrt theta2, while our target is to compute grad L(psi(..), ..) wrt theta1 + grad ... wrt theta2. And the grad wrt theta1 is not that easily computable: we don't have access to phi, so this gradient can be estimated only somehow weirdly.

Common assumption that people make is that the task is 'almost the same': aka grad psi(theta) wrt theta = 0. In the setting of self-play this is just wrong: task changes quite a bit, so this assumption does not hold. what we can do? easy: grad L(psi(theta1), theta2) = grad L wrt arg2 + grad L(psi(theta1)) wrt theta1 = (grad L wrt arg1) * (grad psi wrt theta)  + grad L wrt arg2. Our standard gradient estimators give us the second term, so let's ignore it. grad L wrt arg1 is simply 1 by definition (its not, but whatebver you get the point, L is just a wrapper), so in fact we need to comput grad loss(psi(theta)) wrt theta, where loss is a particular loss on a part of dataset, psi is our prediction of the loss dynamics, and theta our current parameters. Clearly we can train psi by using some information from gradient descent. But the question becomes: how to balance? Probably Bayesian methods could help. All this shit is really hard and i need to much brain to actually solve it without a piece of paper

Somebody, remind me to talk with you about it on the next meeting

**Comment by knyazer** [comment]
[link to a video on topic](https://www.youtube.com/watch?v=g_-gy5lcyqU)
